src/actions/write/buyRecord.ts

```diff
@@ -42,23 +66,20 @@ export const buyRecord = async (
     throw new ContractError(`Caller balance is not defined!`);
   }
 
-  // Check if it includes a valid number of years
-  if (!Number.isInteger(years) || years > MAX_YEARS || years <= 0) {
+  // Additional check if it includes a valid number of years (TODO: this may be set in contract settings)
+  if (years > MAX_YEARS || years <= 0) {
     throw new ContractError(
       'Invalid value for "years". Must be an integer greater than zero and less than the max years',
     );
   }
 
   // list of all active tier ID's
-  const activeTierNumbers = currentTiers.map((k, indx) => indx + 1);
-  if (
-    !Number.isInteger(tierNumber) ||
-    !activeTierNumbers.includes(tierNumber)
-  ) {
+  const activeTierNumbers = currentTiers.map((_, indx) => indx + 1);
+  if (!activeTierNumbers.includes(tierNumber)) {
```

Comment:
This seems like a similar problem as the years issue above. We do a bit less boilerplate checking but still need to check for how values square up against the state/settings from elsewhere.

Reply:
correct - we likely don't want to define the constraints in the schema, unless we are comfortable with evolutions to change them vs setting them in state.

Reply:
We can probably make use of sensible MINIMUMS for most stuff. Maxes is where we might struggle to define durable values.